File: web/client/src/app/research/models/template-section.ts

import { TemplateQuestion } from './template-questions';
import { TemplateSectionGroup } from './template-section-group';

export interface TemplateSection {
    id: number;
    templateId: number;
    sectionId: number;
    name: string;
    description: string;
    hasQuestionGrouping: boolean;
    systemControlTypeId: number;
    orderIndex: number;
    templateSectionGroupId: number;
    templateSectionGroup: TemplateSectionGroup;
    questions: TemplateQuestion[];
}

File: web/api/afmr.model/Research/TemplateSection.cs
using System.Collections.Generic;

namespace afmr.model.Research
{
    public class TemplateSection
    {
        public int Id { get; set; }

        public int TemplateId { get; set; }

        public int SectionId { get; set; }

        public int SubIndex { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public bool HasQuestionGrouping { get; set; }

        public int? SystemControlTypeId { get; set; }

        public bool CanDelete { get; set; }

        public int OrderIndex { get; set; }

        public int? TemplateSectionGroupId { get; set; }

        public TemplateSectionGroup TemplateSectionGroup { get; set; }

        public IEnumerable<TemplateQuestion> Questions { get; set; }
    }
}